app/src/main/java/com/group80/uoftinder/ContactsActivity.java

```diff
@@ -0,0 +1,134 @@
+        FirebaseFirestore firebaseFirestore = FirebaseFirestore.getInstance();
+        String uid = firebaseAuth.getCurrentUser().getUid();
+        Query query = firebaseFirestore.collection("Users").whereArrayContains("contacts", uid);
+        //******************************************************************************************
+
+        FirestoreRecyclerOptions<ContactModel> contacts = new FirestoreRecyclerOptions.Builder<ContactModel>().setQuery(query, ContactModel.class).build();
+        contactAdapter = new FirestoreRecyclerAdapter<ContactModel, ContactViewHolder>(contacts) {
+            @Override
+            protected void onBindViewHolder(@NonNull ContactViewHolder holder, int position, @NonNull ContactModel contactModel) {
+                holder.onBind(getApplicationContext(), position, contactModel);
+            }
+
+            @NonNull
+            @Override
+            public ContactViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+                View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.uoftinder_contact_list_contact_layout, parent, false);
+                return new ContactViewHolder(view);
+            }
+        };
+
+        RecyclerView recyclerView = findViewById(R.id.contactActivityRecyclerView);
+        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this);
+        recyclerView.setHasFixedSize(true);
+        recyclerView.setLayoutManager(linearLayoutManager);
+        recyclerView.setAdapter(contactAdapter);
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        contactAdapter.startListening();
+    }
+
+    @Override
+    public void onStop() {
+        super.onStop();
+        if (contactAdapter != null) {
+            contactAdapter.stopListening();
+        }
+    }
+
+    private class ContactViewHolder extends RecyclerView.ViewHolder {
+        protected final TextView contactName;
+        protected final ImageView contactPic;
+
+        public ContactViewHolder(@NonNull View itemView) {
+            super(itemView);
+            contactName = itemView.findViewById(R.id.contactListContactName);
+            contactPic = itemView.findViewById(R.id.contactListContactPic);
+        }
+
+        /**
+         * Triggered when this View Holder is bind to an adapter
+         * <p>
+         * // TODO: this method may require architectural fix
+         *
+         * @param context      the global information about an application environment // TODO: architectural fix?
+         * @param position     the position of this View Holder in the adapter
+         * @param contactModel a model storing the contact information // TODO: architectural fix
+         */
+        protected void onBind(Context context, int position, @NonNull ContactModel contactModel) {
+            contactName.setText(contactModel.getName());
+
+            // TODO: extract the set-to-image-view process?
```

Comment:
I agree that the set-to-image-view process should be extracted into a method so it's more clear what is going on here

Reply:
Currently I cannot extract this to another method -- considering this logic chunk touches both a use case and a presenter / view, I doubt where it fits in Clean Architecutre. If we need this chunk for the third time in the future then we need to find a way to make it fit into the architecture (this would happen in #60)